src/EditorFeatures/Core/DocumentationComments/DocumentationCommentSuggestion.cs

```diff
@@ -95,6 +108,46 @@ private async Task<bool> TryDisplayProposalAsync(SuggestionSessionBase session,
             return false;
         }
 
+        public async Task DismissSuggestionSessionAsync(CancellationToken cancellationToken)
+        {
+            await DisposeAsync().ConfigureAwait(false);
+            await RunWithEnqueueActionAsync(
+                "DismissSuggestionSession",
+                async () => await _suggestionSession!.DismissAsync(ReasonForDismiss.DismissedDueToInvalidProposal, cancellationToken).ConfigureAwait(false),
+                cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// In general, calls to a SuggestionManager or SuggestionSession need to be wrapped in an EnqueueAction.
+        /// This is the pattern recommended by VS Platform to avoid races.
+        /// </summary>
+        public async Task RunWithEnqueueActionAsync(string description, Func<Task> action, CancellationToken cancel)
+        {
+            Assumes.NotNull(SuggestionManager);
+
+            var taskCompletionSource = new TaskCompletionSource<bool>();
+
+            await providerInstance.ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancel);
+            SuggestionManager.EnqueueAction(description, async () =>
```

Comment:
this is super weird.  they expect you to pass in an async function.  but then they dont' make this function itself task-returning so you can track it?

Reply:
talked offline